test/correctness/align_bounds.cpp

```diff
@@ -197,6 +197,29 @@ int main(int argc, char **argv) {
         }
     }
 
+    // Try a case where aligning a buffer means that strided loads can
+    // do dense aligned loads and then shuffle. This used to trigger a
+    // bug in codegen.
+    {
+        Func f, g;
+        Var x;
+
+        f(x) = x;
+
+        // Do strided loads of every possible alignment
+        Expr e = 0;
+        for (int i = -32; i <= 32; i++) {
+            e += f(3 * x + i);
```

Comment:
I wonder if LLVM might rewrite this to one load plus a bunch of shuffles, which might hide alignment issues? Maybe not now, but it could later?

To avoid this, maybe it would be better to put a loop around the whole thing including the realize, loading one at a time.

Reply:
True, but if the test doesn't reliably test the thing we need testing, it doesn't matter how fast it is. (I guess if this works currently then it's ok, but might be worth capturing this thought in a comment if it breaks again.)

Reply:
I also think the kind of transformation Dillon is talking about is vanishingly unlikely. We're codegenning three consecutive dense vector loads followed by a big shuffle. LLVM isn't going to change that to different loads followed by a different shuffle. The loads are already optimal.

Reply:
If LLVM ever started doing that sort of transformation, we'd need to disable it anyway, because we have scheduling directives to do that and they would stop working.